From 8e23da2f7bf29b841ec89c56a0588e1278d30987 Mon Sep 17 00:00:00 2001
From: Icenowy Zheng <icenowy@aosc.io>
Date: Mon, 1 May 2017 13:58:37 +0800
Subject: [PATCH 77/87] ARM: sun8i: v40: enable SATA on Banana Pi M2 Berry

The Banana Pi M2 Berry board has a SATA port connected to the SATA
controller on V40 SoC. The controller's 1.2v VDD is connected to ELDO3
and 2.5v VDD is connected to DLDO4.

Enable SATA and the needed regulators in the device tree.

Signed-off-by: Icenowy Zheng <icenowy@aosc.io>
---
 arch/arm/boot/dts/sun8i-v40-bananapi-m2-berry.dts | 19 +++++++++++++++++++
 1 file changed, 19 insertions(+)

diff --git a/arch/arm/boot/dts/sun8i-v40-bananapi-m2-berry.dts b/arch/arm/boot/dts/sun8i-v40-bananapi-m2-berry.dts
index 67cf87f3d5c7..8d92b7a5138a 100644
--- a/arch/arm/boot/dts/sun8i-v40-bananapi-m2-berry.dts
+++ b/arch/arm/boot/dts/sun8i-v40-bananapi-m2-berry.dts
@@ -87,6 +87,13 @@
 	};
 };
 
+&ahci {
+	target-supply = <&reg_vcc5v0>;
+	vdd1v2-supply = <&reg_eldo3>;
+	vdd2v5-supply = <&reg_dldo4>;
+	status = "okay";
+};
+
 &ehci1 {
 	status = "okay";
 };
@@ -155,6 +162,18 @@
 	regulator-name = "vcc-wifi";
 };
 
+&reg_dldo4 {
+	regulator-min-microvolt = <2500000>;
+	regulator-max-microvolt = <2500000>;
+	regulator-name = "vdd-2v5-sata";
+};
+
+&reg_eldo3 {
+	regulator-min-microvolt = <1200000>;
+	regulator-max-microvolt = <1200000>;
+	regulator-name = "vdd-1v2-sata";
+};
+
 &mmc0 {
 	vmmc-supply = <&reg_dcdc1>;
 	bus-width = <4>;
-- 
2.13.5

